modules/catalog/src/test/java/org/apache/ignite/internal/catalog/commands/CreateTableCommandValidationTest.java

```diff
@@ -0,0 +1,421 @@
+import org.apache.ignite.internal.catalog.ClockWaiter;
+import org.apache.ignite.internal.catalog.UpdateProducer;
+import org.apache.ignite.internal.catalog.commands.builders.CreateTableCommandBuilder;
+import org.apache.ignite.internal.catalog.descriptors.CatalogHashIndexDescriptor;
+import org.apache.ignite.internal.catalog.descriptors.CatalogIndexDescriptor;
+import org.apache.ignite.internal.catalog.descriptors.CatalogSchemaDescriptor;
+import org.apache.ignite.internal.catalog.descriptors.CatalogTableColumnDescriptor;
+import org.apache.ignite.internal.catalog.descriptors.CatalogTableDescriptor;
+import org.apache.ignite.internal.catalog.descriptors.CatalogZoneDescriptor;
+import org.apache.ignite.internal.catalog.storage.UpdateLog;
+import org.apache.ignite.internal.testframework.BaseIgniteAbstractTest;
+import org.apache.ignite.internal.testframework.IgniteTestUtils.RunnableX;
+import org.hamcrest.Matchers;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+/**
+ * Tests to verify validation of {@link CreateTableCommand}.
+ */
+@SuppressWarnings("DataFlowIssue")
+public class CreateTableCommandValidationTest extends BaseIgniteAbstractTest {
+    private static final String SCHEMA_NAME = "PUBLIC";
+    private static final String ZONE_NAME = "DEFAULT";
+
+    private static final CatalogZoneDescriptor DEFAULT_ZONE = new CatalogZoneDescriptor(
+            0, ZONE_NAME, 1, -1, -1, -1, -1, "", null
+    );
+
+    private final CatalogManager manager = new CatalogManagerImpl(
+            mock(UpdateLog.class),
+            mock(ClockWaiter.class)
+    );
+
+    @Test
+    void schemaNameMustBeNullOrBlank() {
+        CreateTableCommandBuilder builder = manager.createTableCommandBuilder();
+
+        builder = fillProperties(builder);
+
+        for (String name : new  String[] {null, "", " ", "  "}) {
+            builder.schemaName(name);
+
+            assertThrows(
+                    builder::build,
+                    CatalogValidationException.class,
+                    "Name of the schema can't be null or blank"
+            );
+        }
+    }
+
+    @Test
+    void tableNameMustBeNullOrBlank() {
+        CreateTableCommandBuilder builder = manager.createTableCommandBuilder();
+
+        builder = fillProperties(builder);
+
+        for (String name : new  String[] {null, "", " ", "  "}) {
+            builder.tableName(name);
+
+            assertThrows(
+                    builder::build,
+                    CatalogValidationException.class,
+                    "Name of the table can't be null or blank"
+            );
+        }
+    }
+
+    @Test
+    @SuppressWarnings("unchecked")
+    void tableShouldHaveAtLeastOneColumn() {
+        CreateTableCommandBuilder builder = manager.createTableCommandBuilder();
+
+        builder = fillProperties(builder);
+
+        for (List<ColumnParams> columns : new List[] {null, List.of()}) {
```

Comment:
Maybe this test case can also be parameterised with `null, List.of()` and `tableShouldHaveAtLeastOnePrimaryKeyColumn` as well?

Reply:
fixed